### ISSUE TITLE
[Problem]: DSP Convolution is not applied when shairport-sync is running as service

### ISSUE THREAD

**Issue by janhenhan** [mikebrady/shairport-sync#1787]
### What happened?

Hi!

I'm running shairport sync on a RPI 4 under Raspbian. Built it from source.
I built it with options to use the DSP functionality to convolve with some filters that I created. The path to the filter is correctly set in /etc/shairport-sync.conf 
It all works great when I start shairport sync from the command line. The convolution is applied correctly. 
However when I start it as a systemctl service the filter is no longer applied! I can see in the systemctl status of the service that it does seem to use the right config and that the path is still pointing to the right filter. Just the output on my speakers is no longer filtered.

Any advice is appreciated!!
Thanks

### System Information.

RPI 4 under Raspbian. Built it from source.

### Configuration Information.

```shell
RPI 4 under Raspbian. Built it from source.
```

### PulseAudio or PipeWire installed?

- [ ] Check if your system uses a Sound Server.

### How did you install Shairport Sync?

Built from source

### Check previous issues

- [X] Confirm

**Comment by mikebrady** [comment]
Thanks for the post, and apologies for the delay.

This sounds either like a file path specification problem or a file permissions problem. Recall that when Shairport Sync is running as a service, it runs with user and group both set to `shairport-sync`.

This means that any file specification must use the complete file path, and can not assume a default home folder.

The second thing, of course, is that the permissions should be such that Shairport Sync can access and read the file -- maybe make it world-readable.

If you set the log verbosity to `1` and try to run Shairport Sync as a service, messages will be logged in the system log, and they might help you to isolate the problem.

**Comment by janhenhan** [comment]
Thank you very much for your quick reply!

Indeed this was resolved after I copied the file to a world-readable location and made the file world-readable as well.
I already had the log verbosity set to 1 - there is no difference in output whether the path exists and is readable by shairport or whether I point it to a fantasy path. Maybe worth giving some user feedback there :)

Thanks again! Great project!
